sktime/transformations/series/tests/test_imputer.py

```diff
@@ -58,6 +58,18 @@ def test_imputer(method, Z, value, forecaster):
     assert not y_hat.isnull().to_numpy().any()
 
 
+@pytest.mark.parametrize("method", METHODS)
+def test_impute_multiindex(method):
+    """Test for data leakage in case of pd-multiindex data.
+
+    Failure case in bug #6224
+    """
+    df = get_examples(mtype="pd-multiindex")[0]
+    df.iloc[3:6, :] = np.nan  # instance 1 entirely missing
+    df_imp = Imputer(method="ffill").fit_transform(df)
+    assert np.array_equal(df, df_imp, equal_nan=True)
```

Comment:
* do we want to test that no nans are left?
* do we want to add a case where only start or end of an instance are missing? This could be tested in the same example, e.g., one full instance, one completely missing, one at the start, one at the end
* if it is not too much of a hassle, one could also test consistency between applying the imputer to every instance separately, vs applying them to the panel. That would avoid regression of the issue where we impute "across" instances.

Reply:
Great suggestions, added them to the test!